apis/kubernetes/v1alpha1/types.go

```diff
@@ -17,26 +17,27 @@ limitations under the License.
 package v1alpha1
 
 import (
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
 )
 
 // A ProviderSpec defines the desired state of a Provider.
 type ProviderSpec struct {
-	// Kubernetes credentials Secret reference
 	// A Secret containing connection credentials for a Kubernetes cluster
 	// client that will be used to authenticate to this Kubernetes Provider.
 	// This will typically be the connection secret of a KubernetesCluster claim,
 	// or the secret created by a Kubernetes service account, but could also be
 	// manually configured to connect to a preexisting cluster.
-	Secret corev1.LocalObjectReference `json:"credentialsSecretRef"`
+	Secret v1alpha1.SecretKeySelector `json:"credentialsSecretRef"`
```

Comment:
```suggestion
	Secret v1alpha1.SecretReference `json:"credentialsSecretRef"`
```

Reply:
This secret will contain k8s client config data so the key portion of the `SecretKeySelector` will always go unused (i.e. we are interested in multiple keys). ref: https://github.com/crossplaneio/stack-rook/blob/b6df7a99300a77a949ab7b4f18dcd839f0b3b802/pkg/clients/client.go#L33